front/components/assistant/RenderMessageMarkdown.tsx

```diff
@@ -493,16 +453,20 @@ function PreBlock({
         : null;
   }
 
-  const handleCopyPre = async () => {
-    const text = validChildrenContent || fallbackData || "";
-    void copyToClipboard(
-      new ClipboardItem({
-        "text/plain": new Blob([text], {
-          type: "text/plain",
-        }),
-      })
-    );
-  };
+  const text = validChildrenContent || fallbackData || "";
+  const language = detectLanguage(children);
+
+  // If the output file is a CSV let the user download it.
+  const getContentToDownload: GetContentToDownloadFunction | undefined =
```

Comment:
why not let them download other kinds of content?

Reply:
This time will come 🙃